### ISSUE TITLE
store platform & arch information next to installed binary

### ISSUE THREAD

**Issue by derhuerst** [eugeneware/ffmpeg-static#48]
#35 made clear that, in a cross-platform-packaging project using `ffmpeg-static` as a dependency, **a second `npm install` should install an `ffmpeg` binary for a *different platform/arch than before***.

To enable this, `install.js` needs to

1. [ ] write the platform & arch next to the binary (e.g. to `ffmpeg.json`).
1. [ ] check if there's a `ffmpeg.json` in place, with a different platform/arch than desired. If platform & arch are equal, it can exit successfully.

**Comment by snobu** [comment]
I would suggest a naming convention for the ffmpeg binary rather than additional metadata file.

**Comment by derhuerst** [comment]
True, that would work as well. We could follow the Go or Rust binary naming scheme here.

**Comment by terribleplan** [comment]
The benefits I see with something like an `ffmpeg.json` or similar is that it:
1. Allows for for never having to list the directory and have conditional code for the platform as long is it is not changed between build/install time and runtime. (Store `"filename": "ffmpeg.exe"` to make windows just work). Else you would have to list the dir, see what is in it, find something that starts with "ffmpeg-", detect platform and arch by splitting on "-", figure out what to do if there are multiple, etc... 
2. Allows for drift detection. (Store `"hash": "sha256 of the file"`, allows for detecting if something has gotten weird/been modified/etc) Could be an interesting future feature, at least debugging tool somewhat.
